cli/exchange/user.go

```diff
@@ -66,18 +69,26 @@ func UserCreate(org, userPwCreds, user, pw, email string, isAdmin bool, isHubAdm
 	// get message printer
 	msgPrinter := i18n.GetMessagePrinter()
 
-	if email == "" {
-		if strings.Contains(user, "@") {
-			email = user
-		} else {
-			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the email must be specified via -e if the username is not an email address."))
-		}
+	var emailIsOptional bool
+	var err error
+	if emailIsOptional, err = checkExchangeVersionForOptionalUserEmail(org, userPwCreds); err != nil {
+		cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("failed to check exchange version, error: %v", err))
+	}
+
+	if email == "" && strings.Contains(user, "@") {
+		email = user
+	} else if email == "" && !emailIsOptional {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the email must be specified via -e if the username is not an email address."))
```

Comment:
email is a positional parameter, it does not use the -e flag. I think this message should also indicate that email is not optional because the exchange doesnt support it. How about this:
"an email must be specified because the Exchange API version is less than %v.", USER_EMAIL_OPTIONAL_EXCHANGE_VERSION